autoload/slime.vim

```diff
@@ -2,23 +2,54 @@
 " Target Interface
 """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
+" TargetSend tries to send the text and its config to the right place, by
+" finding the first existing variable in this list and performing the
+" corresponding action.
+" 1. b:slime_target_send  -> Trying to execute a `system` call, passing the
+" configuration in a `CONFIG` environment variable,
+" 2. b:SlimeTargetSend    -> Calling the function with `config` and `text`,
+" 3. g:slime_target_send  ->  Trying to execute a `system` call, passing the
+" configuration in a `CONFIG` environment variable,
+" 4. g:SlimeTargetSend    -> Calling the function with `config` and `text`.
 function! s:TargetSend(config, text)
-  let output = system("CONFIG=" . shellescape(a:config) . " " . g:slime_target_send, a:text)
+  if exists("b:slime_target_send")
+    let output = system("CONFIG=" . shellescape(a:config) . " " . b:slime_target_send, a:text)
+  elseif exists("*b:SlimeTargetSend")||exists("b:SlimeTargetSend") 
+    let output = b:SlimeTargetSend(a:config, a:text)
+  elseif exists("g:slime_target_send")
+    let output = system("CONFIG=" . shellescape(a:config) . " " . g:slime_target_send, a:text)
+  elseif exists("*g:SlimeTargetSend")||exists("g:SlimeTargetSend") 
+    let output = g:SlimeTargetSend(a:config, a:text)
+  else
+    echoerr "vim-slime could not determine a valid target !"
+  endif
   if v:shell_error
     echoerr output
   endif
 endfunction
 
+" TargetConfig tries to configure vim-slime. It looks for a function to
+" delegate to and calls the first defined function in this list :
+" 1. b:SlimeTargetConfig
+" 2. g:SlimeTargetConfig
+"
+" If nothing is found, it silently returns an empty string, allowing for
+" no-configuration plugins.
 function! s:TargetConfig() abort
   if exists("b:slime_config")
     return b:slime_config
   end
-  let output = system(g:slime_target_config)
+  if exists("*b:SlimeTargetConfig")||exists("b:SlimeTargetConfig") 
+    let output = b:SlimeTargetConfig()
+  elseif exists("*g:SlimeTargetConfig")||exists("g:SlimeTargetConfig")
+    let output = g:SlimeTargetConfig()
+  else
+    return ""
```

Comment:
is "no config" an error? :thinking:

Reply:
I thought that in some cases you don't really need a config, so no error should be thrown. (see the first example I put in the README.md file)